In [5]:
import sklearn
from nltk.corpus import gutenberg

list the files in Gutenberg corpus

In [6]:
nltk.corpus.gutenberg.fileids()

[u'austen-emma.txt',
 u'austen-persuasion.txt',
 u'austen-sense.txt',
 u'bible-kjv.txt',
 u'blake-poems.txt',
 u'bryant-stories.txt',
 u'burgess-busterbrown.txt',
 u'carroll-alice.txt',
 u'chesterton-ball.txt',
 u'chesterton-brown.txt',
 u'chesterton-thursday.txt',
 u'edgeworth-parents.txt',
 u'melville-moby_dick.txt',
 u'milton-paradise.txt',
 u'shakespeare-caesar.txt',
 u'shakespeare-hamlet.txt',
 u'shakespeare-macbeth.txt',
 u'whitman-leaves.txt']

Lets peek into one of the text: 'shakespeare-hamlet.txt'

In [8]:
hamlet_sentences = gutenberg.sents('shakespeare-hamlet.txt')

so the sentences are already seperated, no need to split them

[u'[', u'The', u'Tragedie', u'of', u'Hamlet', u'by', u'William', u'Shakespeare', u'1599', u']']
[u'Actus', u'Primus', u'.']
[u'Scoena', u'Prima', u'.']
[u'Enter', u'Barnardo', u'and', u'Francisco', u'two', u'Centinels', u'.']
[u'Barnardo', u'.']
[u'Who', u"'", u's', u'there', u'?']
[u'Fran', u'.']
[u'Nay', u'answer', u'me', u':', u'Stand', u'&', u'vnfold', u'your', u'selfe']
[u'Bar', u'.']
[u'Long', u'liue', u'the', u'King']
[u'Fran', u'.']
[u'Barnardo', u'?']
[u'Bar', u'.']
[u'He']
[u'Fran', u'.']
[u'You', u'come', u'most', u'carefully', u'vpon', u'your', u'houre']
[u'Bar', u'.']
[u"'", u'Tis', u'now', u'strook', u'twelue', u',', u'get', u'thee', u'to', u'bed', u'Francisco']
[u'Fran', u'.']
[u'For', u'this', u'releefe', u'much', u'thankes', u':', u"'", u'Tis', u'bitter', u'cold', u',', u'And', u'I', u'am', u'sicke', u'at', u'heart']
[u'Barn', u'.']
[u'Haue', u'you', u'had', u'quiet', u'Guard', u'?']
[u'Fran', u'.']
[u'Not', u'a', u'Mouse', u'stirring']
[u'Barn', u'.']
[u'Well', u',', 

[u'King', u'.']
[u'Welcome', u'deere', u'Rosincrance', u'and', u'Guildensterne', u'.']
[u'Moreouer', u',', u'that', u'we', u'much', u'did', u'long', u'to', u'see', u'you', u',', u'The', u'neede', u'we', u'haue', u'to', u'vse', u'you', u',', u'did', u'prouoke', u'Our', u'hastie', u'sending', u'.']
[u'Something', u'haue', u'you', u'heard', u'Of', u'Hamlets', u'transformation', u':', u'so', u'I', u'call', u'it', u',', u'Since', u'not', u'th', u"'", u'exterior', u',', u'nor', u'the', u'inward', u'man', u'Resembles', u'that', u'it', u'was', u'.']
[u'What', u'it', u'should', u'bee', u'More', u'then', u'his', u'Fathers', u'death', u',', u'that', u'thus', u'hath', u'put', u'him', u'So', u'much', u'from', u'th', u"'", u'vnderstanding', u'of', u'himselfe', u',', u'I', u'cannot', u'deeme', u'of', u'.']
[u'I', u'intreat', u'you', u'both', u',', u'That', u'being', u'of', u'so', u'young', u'dayes', u'brought', u'vp', u'with', u'him', u':', u'And', u'since', u'so', u'Neighbour', u"'", u'd', u'to', u'

In [12]:
len(hamlet_sentences)

3106

lets simply count all words and see if can build a word-document matrix

In [14]:
from sklearn.feature_extraction.text import CountVectorizer 

In [23]:
count_vectorizer = CountVectorizer(tokenizer= lambda i :i, lowercase= False, stop_words= None)
document_term_matrix = count_vectorizer.fit_transform(hamlet_sentences)


In [26]:
print term_document_matrix.shape

(3106, 5447)


we have a matrix of 3106 rows equal to number of sentences and 5447 rows which is the vocabulary size